internal/provider/resource_time_offset.go

```diff
@@ -176,11 +176,110 @@ func (t timeOffsetResourceType) NewResource(ctx context.Context, p tfsdk.Provide
 var (
 	_ tfsdk.Resource                = (*timeOffsetResource)(nil)
 	_ tfsdk.ResourceWithImportState = (*timeOffsetResource)(nil)
+	_ tfsdk.ResourceWithModifyPlan  = (*timeOffsetResource)(nil)
 )
 
 type timeOffsetResource struct {
 }
 
+func (t timeOffsetResource) ModifyPlan(ctx context.Context, req tfsdk.ModifyResourcePlanRequest, resp *tfsdk.ModifyResourcePlanResponse) {
+	// Plan only needs modifying if the resource already exists as the purpose of
+	// the plan modifier is to show updated attribute values on CLI.
+	if req.State.Raw.IsNull() {
+		return
+	}
+
```

Comment:
I think when we were discussing the `ModifyPlan` that I might have skipped over implementing the equivalent for `resourceTimeOffsetConditionExpirationChange()` that's used in the `CustomizeDiff` when running with SDKv2.

Think we'll need something like the following:
```go
	var state, plan timeOffsetModelV0

	diags := req.State.Get(ctx, &state)
	resp.Diagnostics.Append(diags...)
	if resp.Diagnostics.HasError() {
		return
	}

	diags = req.Plan.Get(ctx, &plan)
	resp.Diagnostics.Append(diags...)
	if resp.Diagnostics.HasError() {
		return
	}

	if state.OffsetYears == plan.OffsetYears &&
		state.OffsetMonths == plan.OffsetMonths &&
		state.OffsetDays == plan.OffsetDays &&
		state.OffsetHours == plan.OffsetHours &&
		state.OffsetMinutes == plan.OffsetMinutes &&
		state.OffsetSeconds == plan.OffsetSeconds {
		return
	}
``` 

We'll also need to remove unmarshalling of the plan that currently happens on lines 218-224 as we're now doing that here.